### PR TITLE
test(NODE-4562): reduce macos coverage

### PR DESCRIPTION
**Pull request by baileympearson**
#### What is changing?

This PR updates our CI matrix to only test the rapid release on a standalone server on macos.

##### Is there new documentation needed for these changes?

No.

#### What is the motivation for this change?

Green CI.

### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .evergreen/generate_evergreen_tasks.js [hunk 1]
```diff
@@ -378,6 +374,17 @@ OPERATING_SYSTEMS.forEach(
   }
 );
 
+BUILD_VARIANTS.push({
+  name: 'macos-1014',
```
Comment (durran) [inline .evergreen/generate_evergreen_tasks.js:378]: Just a suggestion to maybe alleviate some perf issues - maybe we try using `macos-1100` here to have the latest? There are 67 allocated hosts on macstadium for evergreen to use in this distro and maybe many other projects haven't updated to that yet?

**Review by durran — CHANGES_REQUESTED** [review]

File: .evergreen/generate_evergreen_tasks.js (hunk 1 above)
Comment (baileympearson) [reply .evergreen/generate_evergreen_tasks.js:378]: Sure, I'll bump it.

**Review by durran — APPROVED** [review]
Looks like a rebase then LGTM.

File: .evergreen/generate_evergreen_tasks.js [hunk 2]
```diff
@@ -378,6 +374,17 @@ OPERATING_SYSTEMS.forEach(
   }
 );
 
+BUILD_VARIANTS.push({
+  name: 'macos-1100',
+  display_name: `macOS 10.14 Node ${LATEST_LTS}`,
```
Comment (durran) [inline .evergreen/generate_evergreen_tasks.js:379]: MacOS 11 or Big Sur

File: .evergreen/generate_evergreen_tasks.js [hunk 3]
```diff
@@ -378,6 +374,17 @@ OPERATING_SYSTEMS.forEach(
...
+  run_on: 'macos-1014',
```
Comment (durran) [inline .evergreen/generate_evergreen_tasks.js:380]: macos-1100

**Review by durran — CHANGES_REQUESTED** [review]

**Review by nbbeeken — COMMENTED** [review]
Do we want to consider replica set just so we can continue testing change streams on mac os? It's one notable area that would be omitted with a standalone only task

**Review by durran — APPROVED** [review]
I guess 11 has the same timeout issue so I guess we just have to expect it on any MacOS.

**Review by durran — APPROVED** [review]

**Review by nbbeeken — APPROVED** [review]

**Pull request merged**